src/04-test-class/index.test.ts

```diff
@@ -1,44 +1,99 @@
 // Uncomment the code below and write your tests
-// import { getBankAccount } from '.';
+import {
+  InsufficientFundsError,
+  SynchronizationFailedError,
+  TransferFailedError,
+  getBankAccount,
+} from '.';
 
 describe('BankAccount', () => {
   test('should create account with initial balance', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    expect(account.getBalance()).toBe(1000);
   });
 
   test('should throw InsufficientFundsError error when withdrawing more than balance', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    const operation = () => account.withdraw(10000);
+    expect(operation).toThrow(new InsufficientFundsError(1000));
   });
 
   test('should throw error when transferring more than balance', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    const accountToTransfer = getBankAccount(0);
+    const operation = () => account.transfer(10000, accountToTransfer);
+    expect(operation).toThrow(new InsufficientFundsError(1000));
   });
 
   test('should throw error when transferring to the same account', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    const operation = () => account.transfer(10000, account);
+    expect(operation).toThrow(new TransferFailedError());
   });
 
   test('should deposit money', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    account.deposit(1000);
+    expect(account.getBalance()).toBe(2000);
   });
 
   test('should withdraw money', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    expect(account.withdraw(100).getBalance()).toBe(900);
   });
 
   test('should transfer money', () => {
-    // Write your test here
+    const account = getBankAccount(1000);
+    const accountToTransfer = getBankAccount(0);
+    account.transfer(1000, accountToTransfer);
+    expect(account.getBalance()).toBe(0);
+    expect(accountToTransfer.getBalance()).toBe(1000);
   });
 
   test('fetchBalance should return number in case if request did not failed', async () => {
-    // Write your tests here
+    expect.assertions(1);
+    const account = getBankAccount(1000);
+    const operation = await account.fetchBalance();
+    if (operation === null) {
+      expect(operation).toBe(null);
```

Comment:
There's no sense in asserting `null` if it was already known to be `null`.
The idea is to mock `lodash.random` in the way `fetchBalance` always returns number (actual balance).